### DISCUSSION TITLE
issue: ZodResolver doesn't update form error state with multiple fields involved

### DISCUSSION THREAD

**Discussion by BaileyMillerSSI** [react-hook-form/react-hook-form#10745] [Q&A] [answered]
### Version Number

^7.43.9

### Codesandbox/Expo snack

https://stackblitz.com/edit/stackblitz-starters-rnayem?file=src%2FExampleForm.tsx

### Steps to reproduce

1. Go to https://stackblitz.com/edit/stackblitz-starters-rnayem?file=src%2FExampleForm.tsx
2. Start from the ExampleForm file, notice the schema as defined by Zod. 
3. Look for the superRefine section of the schema. Notice that within that super refine I am watching 4 fields and adding an error to the ticket price and guest ticket price when they have a value but the check boxes aren't selected. 
4. Within the application enter any ticket price and any guest price, notice the errors appear. Then select either of the checkboxes and notice the errors doesn't disappear. However, notice that the submit button's validation state has updated but the errors still persist. (The form is truly valid at this point but the fieldState.errors haven't updated)

### Expected behaviour

I would like the error state to update correctly within the (TextInput component) when the form state elsewhere has caused the form to become valid. I am not entirely sure which library is at fault here but it feels like React Hook Form isn't updating the error state correctly when an outside component is the one that triggers the new state.

### What browsers are you seeing the problem on?

Chrome, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
React hook form focus on field level validation, if you need dependent update use trigger api.

**Comment by BaileyMillerSSI** [comment]
For reference:
https://react-hook-form.com/docs/useform/trigger

@bluebill1049 Thanks I'll take a look and potentially post an update to this issue.

**Reply by BaileyMillerSSI** [reply, marked as answer]
Updated example using trigger, which I think solves my problem:
https://stackblitz.com/edit/stackblitz-starters-lrau1a?file=src%2FExampleForm.tsx
